cloud/pkg/devicecontroller/config/kube.go

```diff
@@ -33,6 +35,11 @@ func InitKubeConfig() {
 	if kc, err := config.CONFIG.GetValue("devicecontroller.kube.kubeconfig").ToString(); err != nil {
 		klog.Error("Devicecontroller kube config not set")
 	} else {
+		if strings.HasPrefix(kc, "~") {
+			if homeDir, err := os.UserHomeDir(); err == nil {
+				kc = strings.Replace(kc, "~", homeDir, 1)
```

Comment:
please use '~/' instead of '~' ; it is possible to enter an other user home directory by using '~otherUser'